src/Signer/Hmac.php

```diff
@@ -7,11 +7,18 @@
 
 use function hash_equals;
 use function hash_hmac;
+use function strlen;
 
 abstract class Hmac implements Signer
 {
     final public function sign(string $payload, Key $key): string
     {
+        $actualKeyLength   = strlen($key->contents());
```

Comment:
Should we worry about `mbstring.func_overload` and use `mb_string($key->contents(), '8bit')` instead? That could affect the number of returned bytes when dealing with multibyte chars.

AFAIK the worst-case scenario would be reporting fewer bytes (considering `UTF-8` as the internal default encoding). @Ocramius @Slamdunk do you know of any other implication?

Reply:
From https://www.php.net/manual/en/mbstring.overload.php:

> This feature has been DEPRECATED as of PHP 7.2.0, and REMOVED as of PHP 8.0.0. Relying on this feature is highly discouraged.

Although not scientific, above statement and https://3v4l.org/KHYVk#v8.1.6 are enough for me

Reply:
I know the differences... I just know that not everyone uses random binary strings as keys as they should 😢 (some people are crazy enough to use emojis as variable names 🤣) 

I'm not asking for changes, though, just trying to find out the impact.

Reply:
emojis are a subset of `random_bytes` as far as I know

Reply:
Yeah, the length will be reported differently

Reply:
> Yeah, the length will be reported differently

???

https://3v4l.org/bfP4f#v8.1.6

Reply:
I think you're not comparing it correctly doing is wrong, as the override is likely to be disabled for 3v4l.org.

I've added a link on the previous comment.

Reply:
And the link was wrong 🤦 https://3v4l.org/CTY0s

Reply:
So, as I mentioned, the only implications I can think of is having `strlen()` reporting fewer characters when:

1. the key uses multibyte chars
1. the override is enabled
1. the internal encoding

This is not a security problem as our verification is stricter. Is there any other implication?

Reply:
@Slamdunk this was the only pending discussion but I think we're good as is.
Thanks for your help.